src/Microsoft.AspNet.Http/DefaultHttpContext.cs

```diff
@@ -49,15 +50,33 @@ void IFeatureCache.CheckFeaturesRevision()
         {
             if (_cachedFeaturesRevision !=_features.Revision)
             {
-                _items = null;
-                _serviceProviders = null;
-                _authentication = null;
-                _lifetime = null;
-                _session = null;
-                _cachedFeaturesRevision = _features.Revision;
+                ResetFeatures();
             }
         }
 
+        public void UpdateFeatures(IFeatureCollection features)
+        {
+            _features = features;
+            ResetFeatures();
+
+            _request.UpdateFeatures(features);
+            _response.UpdateFeatures(features);
+
+            _authenticationManager?.UpdateFeatures(features);
+            _connection?.UpdateFeatures(features);
+            _websockets?.UpdateFeatures(features);
```

Comment:
Paid the price; fairly small, are a defined size (rather than an unknown user code type) - might as well? (Happy either way)